### PR TITLE
[Trivial] Fix Issue 17540 - setting CurlOption.noproxy is cumbersome 

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: std/net/curl.d [hunk 1]
```diff
@@ -2048,6 +2048,18 @@ private mixin template Protocol()
     }
 
     /**
+       Set the no proxy flag for the specified host names.
+       Params:
+       test = a list of comma separated list of host names that
+              do not require proxy to get reached
+    */
+
+    @property void setNoProxy(string hosts)
```
Comment (MetaLang) [inline std/net/curl.d]: `@property` is not necessary, especially for a setter function.

File: std/net/curl.d [hunk 2]
```diff
@@ -2048,6 +2048,18 @@ private mixin template Protocol()
     }
 
     /**
+       Set the no proxy flag for the specified host names.
+       Params:
+       test = a list of comma separated list of host names that
+              do not require proxy to get reached
+    */
+
```
Comment (andralex) [inline std/net/curl.d]: No space between ddoc and definition

File: std/net/curl.d [hunk 3]
```diff
@@ -2122,6 +2134,9 @@ private mixin template Protocol()
         http.onReceive = (ubyte[] data) { return data.length; };
         http.setAuthentication("user", "pass");
         http.perform();
+
+        // Bugzilla 17540
```
Comment (andralex) [inline std/net/curl.d]: Remove this, it's irrelevant for the doc reader.

Comment (CyberShadow) [inline std/net/curl.d]: This is not a documented unittest, and referencing bugzilla numbers in tests is generally useful, so it's fine as it is.

Comment (andralex) [inline std/net/curl.d]: Ah, missed that. Thx.

**Comment by RazvanN7** [comment]
@CyberShadow any ideas on why the tester fails?

**Comment by wilzbach** [comment]
(squashed and rebased)

**Pull request merged**